internal/cachecontroller/cache_controller.go

```diff
@@ -100,15 +103,29 @@ func (c *InMemoryCacheController) DetermineInvalidation(
 		return entry.LastModified
 	}
 
-	lastModified, err, _ := c.sf.Do(storeID, func() (interface{}, error) {
-		return c.findChangesAndInvalidate(ctx, storeID)
-	})
-
-	if err != nil {
-		return time.Time{}
+	c.mu.Lock()
+	_, present := c.inflightInvalidations[storeID]
```

Comment:
https://pkg.go.dev/sync#Map

> The [sync.]Map type is optimized for two common use cases: 
> 1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or 
> (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. 
> In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex

I feel that we fall under the (1) bucket, therefore we could replace the map+mutex with a `sync.map`, but i'll defer to you.

Reply:
I don't think it quite fits in this particular bucket because we will keep writing/deleting over the key in its lifetime.  Thus, I have opted for the plain old mutex.

Reply:
@adriantam it is achievable through `LoadOrStore` and `LoadAndDelete`. There is a _minor_ gain there. The same key is only written then deleted, never modified. If we needed to modify it would still work with the Map.

Reply:
updated